crates/bdk/src/wallet/mod.rs

```diff
@@ -371,13 +369,11 @@ impl<L> std::error::Error for LoadError<L> where L: core::fmt::Display + core::f
 /// [`new_or_load`]: Wallet::new_or_load
 /// [`new_or_load_with_genesis_hash`]: Wallet::new_or_load_with_genesis_hash
 #[derive(Debug)]
-pub enum NewOrLoadError<W, L> {
+pub enum NewOrLoadError {
     /// There is a problem with the passed-in descriptor.
     Descriptor(crate::descriptor::DescriptorError),
     /// Writing to the persistence backend failed.
-    Write(W),
-    /// Loading from the persistence backend failed.
-    Load(L),
+    Persist(anyhow::Error),
```

Comment:
We might need to update the docs as this variant can represent either a writing or loading error now.

Reply:
@LLFourn what are your thoughts about combining write/load variants?

Reply:
Fine to keep them as one variant but as you said doc strings need to be updated if so. I don't see a strong motivation to merge them.

Reply:
For the comment above the variant, how is "Either writing to or loading from the persistence backend failed." Verbatim of the `NewOrLoadError::Persist` error message.